codebase/codebase.go

```diff
@@ -37,22 +40,41 @@ func (c *CodebaseContent) ToMap() map[string]interface{} {
 	res[BranchKey] = c.Branch
 	res[FileNameKey] = c.FileName
 	res[LineNumberKey] = c.LineNumber
+	res[CodebaseIDKey] = c.CodebaseID
 	return res
 }
 
+// validateURL makes sure Repo is valid GIT URL
+func validateURL(repo string) (string, bool) {
+	validPrefixes := []string{"https://github.com/", "git@github.com:"}
+	if strings.HasPrefix(repo, validPrefixes[0]) || strings.HasPrefix(repo, validPrefixes[1]) {
+		if strings.HasSuffix(repo, ".git") == false {
+			return repo + ".git", true
```

Comment:
Thank you @hectorj2f, you are right user should give us the valid URL. But currently this information is coming analyzer(no user interaction) hence added validation. In case of `workspace` UI page, we perform URL validations on frontend. But for analyzer I am not sure if URL is validated or not.

About the method, If method name was `IsValid` then returning only `bool` is perfect.
Do you suggest to have two functions ? one to append ".git" and one `isValid()` ?